File: leo/week2/F-2828.js
/**
 *  제목 :
 *  입력 :
 *  제한사항 :
 *  시간계산 필요 :
 */

// Run by Node.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let N, M, J;
let positions = [];
let index = 0;

rl.on("line", function (line) {
  if (index === 0) {
    [N, M] = line.split(" ").map(Number);
    index++;
  } else if (index === 1) {
    J = parseInt(line);
    index++;
  } else {
    positions.push(parseInt(line));
    if (positions.length === J) {
      rl.close();
    }
  }
}).on("close", function () {
  let left = 1; // 바구니의 시작 위치 (1-based index)
  let right = M; // 바구니의 끝 위치
  let moveCount = 0;

  for (let i = 0; i < J; i++) {
    const apple = positions[i];

    if (apple < left) {
      // 사과가 바구니의 왼쪽에 떨어진 경우
      moveCount += left - apple;
      right -= left - apple;
      left = apple;
    } else if (apple > right) {
      // 사과가 바구니의 오른쪽에 떨어진 경우
      moveCount += apple - right;
      left += apple - right;
      right = apple;
    }
    // 사과가 바구니 범위 내에 있는 경우에는 이동하지 않음
  }

  console.log(moveCount);
  process.exit();
});

File: mindy/week2/F-2828.py
# continue : 하위 코딩을 건너뛰고 다음 순번의 loop를 수행한다.
# pass : 실행할 코드가 없는 것으로 다음 행동을 수행한다.
# break : 반복문을 멈추고 loop 밖으로 나간다

from collections import deque
# 입력받기

n, m = map(int, input().split())

j = int(input())

left=1
right = m
cnt = 0


for _ in range(j):
    apple = int(input())
    
    # 사과의 위치가 바구니 내에 있는 경우
    if left <= apple and right>=apple:
        continue
    # 사과의 위치가 바구니보다 왼쪽에 있는 경우
    elif left > apple:
        cnt += (left-apple)
        right -= (left-apple)
        left = apple
    # 사과의 위치가 바구니보다 오른쪽에 있는 경우
    else:
        cnt += (apple - right)
        left += (apple - right)
        right = apple
        
    
print(cnt)


